### PR TITLE
fix(weave): support $size in ORM

### PR DESCRIPTION
**Pull request by jtschoonhoven**
## Summary

Adds a Mongo-style `$size` operand to the trace-server Query DSL. This lets clients express non-empty structured fields with normal Query composition. 

This is required by https://github.com/wandb/core/pull/48318 which needs a way to query for empty maps.

### CODE REVIEW

File: weave/trace_server/orm.py [hunk 1]
```diff
@@ -373,7 +373,6 @@ def prepare(
                 datetime_columns=self.datetime_columns,
             )
             conditions.extend(query_conds)
-
```
Comment (gtarpenning) [inline weave/trace_server/orm.py:376]: 🔪

File: weave/trace_server/interface/query.py [hunk 2]
```diff
@@ -348,11 +363,15 @@ class ContainsSpec(BaseModel):
...
 )
-Operand = LiteralOperation | GetFieldOperator | ConvertOperation | Operation
+Operand = (
+    LiteralOperation | GetFieldOperator | ConvertOperation | SizeOperation | Operation
+)
 
 # Update the models to include the recursive types
 LiteralOperation.model_rebuild()
 GetFieldOperator.model_rebuild()
+ConvertOperation.model_rebuild()
```
Comment (gtarpenning) [inline weave/trace_server/interface/query.py]: interesting that we need this?

File: tests/trace_server/query_builder/test_agent_query_compiler.py [hunk 3]
```diff
@@ -167,6 +167,18 @@ def test_custom_attr_rejected_field_vs_field(self) -> None:
 
 
 class TestOperatorShapes:
+    def test_size_composes_with_comparison(self) -> None:
```
Comment (gtarpenning) [inline tests/trace_server/query_builder/test_agent_query_compiler.py:170]: rather presumptuous to put your test first ;)

**Review by gtarpenning — APPROVED** [review]
lgtm

File: tests/trace/test_feedback.py [hunk 4]
```diff
@@ -2328,3 +2328,71 @@ def test_feedback_query_returns_tz_aware_created_at(client: WeaveClient) -> None
...
+    rated_id = create_feedback("rated", ratings={"_rating_": 0.8})
+
+    all_feedback = client.server.feedback_query(
+        tsi.FeedbackQueryReq(project_id=project_id, fields=["id"])
+    )
+    scored_feedback = client.server.feedback_query(
+        tsi.FeedbackQueryReq(
+            project_id=project_id,
+            fields=["id"],
+            query=tsi.Query.model_validate(
```
Comment (gtarpenning) [inline tests/trace/test_feedback.py:2368]: maybe one more test that uses a different literal value? (negative, float, non-sensical value, etc...)

File: weave/trace_server/interface/query.py (hunk 2 above)
Comment (jtschoonhoven) [reply weave/trace_server/interface/query.py]: We don't. Removed.

File: tests/trace_server/query_builder/test_agent_query_compiler.py (hunk 3 above)
Comment (jtschoonhoven) [reply tests/trace_server/query_builder/test_agent_query_compiler.py:170]: I've been to nice to my agent lately and it's getting entitled. I'll deal with it.
